Campaign user server

nguyenaiden opened the pull request:
-Add route ` POST /:id/campaigns`
-Add Controller and Service to handle adding to joined table campaigns_users
-Add Unit tests to accommodate for added feature
-Add error message based on DB error codes (need a second look on this)
  PosgreSQL error code table`https://www.postgresql.org/docs/9.6/static/errcodes-appendix.html`
-Seed file works for campaigns_users table (may need a second pair of eyes on this)

Commit:
Add basic view for contact lists along with container

Commit:
File rename

Commit:
Add contact list list view component test
Adjusted admin button group nav to accomodate additional contact list view

Commit:
Renamed test file to admin_contactlist + finished action test

Commit:
Finished reducer test

Commit:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into contact-list-view-test

Commit:
Removed add contact list view

Commit:
Altered test language to be more explicit

Commit:
Test language hotfix

Commit:
Add button to add contact list

Commit:
Add go back button on View script page + 1 additional test

Commit:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into button-addition

Commit:
Get all users route, service, controller added at /users

Commit:
Add test to retrieve all users

Commit:
Expand updateUserById service to allow user manangement
Create separate controller
Add route as a PATCH '/users/:id/manage'

Commit:
add corresponding tests for expanded service

Commit:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into manage-users

Commit:
[hot fix] removed comma in test

nguyenaiden committed:
Merge branch 'master' into manage-users

Commit:
Handle non-accepted statuses (no truthy is_banned  and truthy is_active allowed)
Add tests to accommodate new changes

nguyenaiden committed:
Merge branch 'master' into manage-users

Commit:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into manage-users

Commit:
Merge branch 'manage-users' of https://github.com/Twilio-org/phonebank into manage-users

Commit:
Fixed linting issue

Commit:
Removed isAdmin and isBanned check to silently deactivate user if banned.
Removed if key exists check in private helper function addIfExists

Commit:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into campaign-user-server

Commit:
Expand User service to add campaign to user (campaigns_users join table)

Commit:
Add controller, service, and route for addCampaignToUser

Commit:
Merge branch 'master' of https://github.com/Twilio-org/phonebank into campaign-user-server

Commit:
Adjusted seed file to accommodate for changes

Commit:
Add test for user service expansion to handle addCampaignToUser
Seed file is functional

Commit:
Removed trailing spaces

Commit:
Removed unneeded test parameters

allieglotfelty commented on test/server/services/user.js:
```diff
@@ -65,6 +68,66 @@ describe('User service tests', () => {
+        description: 'Script Description 4'
+      };
+
+      this.campaignUserParam = {
```
I'd use the user and campaign ids you're adding as a part of the tests here rather than hard-coding in 3 - could get tricky if other users get added in or taken out of testing down the line.

nguyenaiden replied:
I'm going to have to store the ID somewhere as it comes back when I add the script.

Commit:
Revised error handling on controller for different kinds of errors
Adjusted test to keep track of correct IDs for campaign, contact list, user and script

allieglotfelty commented on test/server/services/user.js:
```diff
@@ -65,6 +68,65 @@ describe('User service tests', () => {
+    it('should add Campaign User association', (done) => {
+      User.addCampaignToUser(this.campaignUserParam)
+        .then((campaignUser) => {
+          expect(campaignUser.models[0].attributes.user_id).to.equal(3);
```
Instead of 3 here, I'd write this.campaignUserParam.id and this.campaignUserParam.campaign_id

Commit:
Removed hard coded ID's

JadziaHax reviewed (approved):
👍
